src/plugins/intel_gpu/src/graph/condition.cpp

```diff
@@ -20,38 +53,83 @@ GPU_DEFINE_PRIMITIVE_TYPE_ID(condition)
+
+    auto layouts_true  = get_output_layouts(get_out_layout_map(impl_param.inner_progs[idx_branch_true]),  impl_param.io_output_maps[idx_branch_true]);
+    auto layouts_false = get_output_layouts(get_out_layout_map(impl_param.inner_progs[idx_branch_false]), impl_param.io_output_maps[idx_branch_false]);
+
     CLDNN_ERROR_LAYOUT_MISMATCH(impl_param.desc->id,
                                 "Branch true output layout",
-                                layout_true,
+                                layouts_true[0],
                                 "branch false output layout",
-                                layout_false,
+                                layouts_false[0],
                                 "Layout of the branches should be the same.");
 
-    return layout_true;
+    return layouts_true[0];
+}
+
+template <class T>
+static bool convert_data(memory::ptr mem, stream& stream) {
+    mem_lock<T, mem_lock_type::read> lock_data{mem, stream};
+    return (static_cast<float>(*lock_data.data()) != 0.f);
+}
+
+bool condition_inst::get_pred_from_memory(memory::ptr mem, stream& stream) {
+    auto mem_dt = mem->get_layout().data_type;
+    switch (mem_dt) {
+        case cldnn::data_types::f32:
+            return convert_data<float>(mem, stream);
+        case cldnn::data_types::f16:
+            return convert_data<half_t>(mem, stream);
+        case cldnn::data_types::i64:
+            return convert_data<int64_t>(mem, stream);
+        case cldnn::data_types::i32:
+            return convert_data<int32_t>(mem, stream);
+        case cldnn::data_types::i8:
+            return convert_data<int8_t>(mem, stream);
+        case cldnn::data_types::u8:
+            return convert_data<uint8_t>(mem, stream);
+        case cldnn::data_types::bin:
+        default:
+            return convert_data<uint32_t>(mem, stream);
+    }
 }
 
+template<typename ShapeType>
+std::vector<layout> condition_inst::calc_output_layouts(condition_node const& node, kernel_impl_params const& impl_param) {
+    if (impl_param.inner_nets.empty()) {
+        OPENVINO_ASSERT(impl_param.inner_progs.empty() == false, "The count of inner programs should not be zero");
+        auto layouts_true  = get_output_layouts(get_out_layout_map(impl_param.inner_progs[idx_branch_true]),  impl_param.io_output_maps[idx_branch_true]);
+        auto layouts_false = get_output_layouts(get_out_layout_map(impl_param.inner_progs[idx_branch_false]), impl_param.io_output_maps[idx_branch_false]);
+
+        if (layouts_true[0].is_static() && layouts_true[0] == layouts_false[0]) {
+            return {layouts_true[0]};
+        } else {
+            OPENVINO_ASSERT(layouts_true[0].get_rank() == layouts_false[0].get_rank(), "dynamic rank is not supported");
+            return {layout{ov::PartialShape::dynamic(layouts_true[0].get_rank()), layouts_true[0].data_type, layouts_true[0].format }};
```

Comment:
As I can see, ngraph shape infer has a bit more complex logic for shape inference

Reply:
I tried to find shape_infer for ngraph::If operator but I can't find it. So I understood that it calculated output layout size in ov::op::v8::If::validate_and_infer_types() without shape_infer function. 
But, If you let me know what I'm missing where ngraph::If's shape_infer function is, I will update it. thanks.

Reply:
https://github.com/openvinotoolkit/openvino/blob/master/src/core/src/op/if.cpp#L59

Reply:
@vladimir-paramuzov , Do you means that calculate output using resolve_shape like shape_infer?

Reply:
as I can see, resolve_shape is internal method and not exposed in headers, so you can't use it. My point is that resolve_shape does a bit more carful shapes handling, for example it keeps static dimensions intact if shape is partially static and tries to merge ranges for each dynamic dimension which may be useful for us too

Reply:
update it according to your comments. please review it again. thanks.